goal_src/jak2/levels/drill_platform/drill-mech-master.gc

```diff
@@ -5,5 +5,434 @@
+          (vector-normalize! (-> s4-0 vector 1) 1.0)
+          (set! (-> s4-0 vector 0 w) 0.0)
+          (set! (-> s4-0 vector 1 w) 0.0)
+          (set! (-> s4-0 vector 2 w) 0.0)
+          (set! (-> s4-0 trans w) 1.0)
+          (spawn-with-matrix (-> self part-doom) s4-0)
+          )
+        (set! (-> self state-time) (-> self clock frame-counter))
+        (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 0.2))
+          (when (and (not gp-1) (>= 2 s5-1) (= (get-status *gui-control* (-> self explosion-sound-id)) (gui-status ready)))
+            (set! gp-1 #t)
+            (set-action!
+              *gui-control*
+              (gui-action play)
+              (-> self explosion-sound-id)
+              (gui-channel none)
+              (gui-action none)
+              (the-as string #f)
+              (the-as (function gui-connection symbol) #f)
+              (the-as process #f)
+              )
+            )
+          (suspend)
+          )
+        )
+      )
+    (sleep-code)
+    (none)
+    )
+  )
+
+(defstate victory (drill-mech-master)
+  :virtual #t
+  :event drill-mech-master-event-handler
+  :code (behavior ()
+    (local-vars (v1-16 process))
+    (remove-setting! 'music)
+    (send-event (handle->process (-> self hud-timer)) 'hide-and-die)
+    (sound-stop (-> self alarm-sound-id))
+    (set! (-> self scene-player-handle)
+          (ppointer->handle (process-spawn scene-player :init scene-player-init "drill-top-explode" #t #f))
+          )
+    (until (not v1-16)
+      (suspend)
+      (set! v1-16 (handle->process (-> self scene-player-handle)))
+      )
+    (process-release? *target*)
+    (sleep-code)
+    (none)
+    )
+  )
+
+;; WARN: Return type mismatch object vs none.
+(defbehavior drill-mech-master-init-by-other drill-mech-master ()
+  (set! *drill-mech-master* (the-as (pointer drill-mech-master) (process->ppointer self)))
+  (set! (-> self exited?) #f)
+  (set! (-> self killed-jak?) #f)
+  (set! (-> self cpanel-count) 10)
+  (set! (-> self hud-cpanel) (the-as handle #f))
+  (set! (-> self hud-timer) (the-as handle #f))
+  (set! (-> self scene-player-handle) (the-as handle #f))
+  (set! (-> self alarm-sound-id) (new 'static 'sound-id))
+  (set! (-> self explosion-sound-id) (new 'static 'sound-id))
+  (set! (-> self total-countdown-time) (seconds 95))
+  (set! (-> self part-doom) (create-launch-control (-> *part-group-id-table* 443) self))
+  (dotimes (gp-0 (-> *level* length))
+    (let ((v1-9 (-> *level* level gp-0)))
+      (when (= (-> v1-9 status) 'active)
+        (let ((s5-0 (-> v1-9 bsp actors)))
+          (when (nonzero? s5-0)
+            (dotimes (s4-0 (-> s5-0 length))
+              (let ((s3-0 (-> s5-0 data s4-0 actor)))
+                (if (and (type-type? (-> s3-0 etype) drill-control-panel) (logtest? (-> s3-0 kill-mask) (task-mask task6)))
+                    (add-icon! *minimap* self (the-as uint 16) (the-as int #f) (the-as vector s3-0) 0)
```

Comment:
this `add-icon!` function is really strange, I've seen it take so many things that aren't an `int` or `vector`